created the HTML file for Student Result page (teacher's view)

Jereriviel opened the pull request:
### What does this PR do?

Adds the page teacher-student-results.html to the templates folder.

As requested in: https://github.com/NoroffFEU/edu-gate/issues/182

### Description of Task to be completed?

Look through code.

### How should this be manually tested?

Can be tested in a browser and validated with the W3C Markup Validation Service.

### Any background context you want to provide?

Jereriviel committed:
created the HTML file for Student Result page (teacher's view)

sulenchy commented on templates/teacher-student-results.html:
```diff
@@ -0,0 +1,248 @@
+      href="https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&family=Saira:ital,wght@0,100..900;1,100..900&display=swap"
+      rel="stylesheet"
+    />
+    <link rel="stylesheet" href="/css/style.css" />
```
You need to link to `'./dist/output.css`

sulenchy commented on templates/teacher-student-results.html:
```diff
@@ -0,0 +1,248 @@
+  </head>
+  <body>
+    <header></header>
+    <main>
```
You need to style them

Jereriviel replied:
Styling is in a different task.

sulenchy commented on templates/teacher-student-results.html:
```diff
@@ -0,0 +1,248 @@
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&family=Saira:ital,wght@0,100..900;1,100..900&display=swap"
```
This is a duplicate

sulenchy commented on templates/teacher-student-results.html:
```diff
@@ -0,0 +1,248 @@
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&family=Saira:ital,wght@0,100..900;1,100..900&display=swap"
```
This does not follow best practices. I recommend you import it in the css/styles.css so that we don't need to import it other HTML files.

Jereriviel replied:
Removed the link to material symbols.

sulenchy reviewed (changes requested):
See comments above

Jereriviel committed:
Edited file after code review

- Added link to output.css
- Removed google link duplicates
- Removed google link Material Symbols

Jereriviel commented:
@sulenchy Thank you for the feedback, I have made changes and the file should be ready now. :)

pauiws reviewed (changes requested):
STRENGTHS: 
-**Good Structure**: Semantic HTML is used appropriately. Logical layout with a table for student results, pagination, and overlays for editing and deleting. 
-**Design-Friendly**: Includes external stylesheets and Google Fonts for styling. Interactive elements like buttons and overlays enhance user experience. 
-**Accessibility and Responsivenes**s: Uses _meta viewport_ for mobile compatibility. Basic accessibility features like placeholders in inputs are present. 

AREAS TO IMPROVE: 
Add _alt_ text for images to support screen readers.

Jereriviel committed:
Added alt attrubutes to img tags

Jereriviel commented:
@pauiws Thank you for the feedback, the changes are completed and the file is ready for review. :)

sulenchy commented:
@kristinatonnessen Please, let's have a ticket for styling this page.

kristinatonnessen commented:
> @kristinatonnessen Please, let's have a ticket for styling this page.

@sulenchy the issue is #309, marcos has already gone over this and added CSS in a new branch, i changed direction for merge so his merge will go into this branch

kristinatonnessen committed:
Merge branch 'develop' into page/student_results_task_182
